json.h

```diff
@@ -81,9 +84,14 @@ static inline int json_object_add_value_string(struct json_object *obj,
 	struct json_value arg = {
 		.type = JSON_TYPE_STRING,
 	};
+	int res;
+	char *string;
 
-	arg.string = strdup(val ? : "");
-	return json_object_add_value_type(obj, name, &arg);
+	string = arg.string = strdup(val ? : "");
+	res = json_object_add_value_type(obj, name, &arg);
+	free(string);
+
+	return res;
 }
```

Comment:
A question...

If the lifetime of `string` is just as short as the `json_object_add_value_type` call, why do we even need `strdup` here? Can't we just do this:
```
arg.string = val ? : "";
return json_object_add_value_type(obj, name, &arg);
```

Reply:
Does `json_object_add_value_type` write to `string[n]`?

Reply:
> Does `json_object_add_value_type` write to `string[n]`?

If not, maybe cast away the const?